config/clusters/cloudbank/ccsf.values.yaml

```diff
@@ -39,22 +39,18 @@ jupyterhub:
           http://google.com/accounts/o8/id:
             username_derivation:
               username_claim: "email"
-            # allow_all is a partial authorization, username_pattern is enforced also
-            allow_all: true
-          urn:mace:incommon:berkeley.edu:
-            username_derivation:
-              username_claim: "email"
-            # allow_all is a partial authorization, username_pattern is enforced also
-            allow_all: true
+            allowed_domains:
+              - mail.ccsf.edu
       Authenticator:
+        allowed_users:
+          - clare.alice.heimer@gmail.com
         admin_users:
           - ericvd@berkeley.edu
           - sean.smorris@berkeley.edu
           - shawn.wiggins@mail.ccsf.edu
           - craig.persiko@mail.ccsf.edu
           - efuchs@mail.ccsf.edu
           - amy.mclanahan@mail.ccsf.edu
-        username_pattern: '^(.+@2i2c\.org|.+@berkeley\.edu|.+@mail\.ccsf\.edu|clare\.alice\.heimer@gmail\.com|deployment-service-check)$'
```

Comment:
Thanks to being able to allow users either by `allowed_domains` and `allowed_users`, we can now grant an individual user access without introducing the more hacky way of configuring `username_pattern`.

Previously, `allowed_domains` once configured would require us to also allow `gmail.com`, but now we can configure `allowed_domains` to authorize a group of users and then surgically adding an individual user via `allowed_users`.

This is the same mechanism that allows us to reduce something like `allowed_domains: ["2i2c.org", "berkeley.edu", "peralta.edu"]` to just `allowed_domains: ["peralta.edu"]` in other hub configurations.

Reply:
ok got it